var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -267,6 +267,15 @@ def extend_config_flags(self):
         classic_flags = ["-gcc-name={}".format(self.compiler.cc)]
         classic_flags.append("-gxx-name={}".format(self.compiler.cxx))
 
+        # `self.compiler.cc` is gcc, so we can assume `-print-multiarch` is a valid option. It is
+        # necessary to explicitly add this directory in cases where the triplet reported by
+        # `gcc -### 2>&1 | grep Target:` does not equal multiarch dir /usr/include/<triplet>.
+        gcc = Executable(self.compiler.cc)
+        multiarch_dir = gcc("-print-multiarch", output=str).strip()
+        triplet = gcc("-###", error=str).partition("Target: ")[2].split("\n")[0]
+        if multiarch_dir and multiarch_dir != triplet:
+            classic_flags.append("-isystem/usr/include/" + multiarch_dir)
```

Comment:
`/usr/include` isn't correct on nix, gentoo prefix, etc.

Reply:
Yes, I was wondering about. How do we get that prefix on other systems?

Reply:
It would be better to find the correct prefix, but my understanding is that it wouldn't be a hard error if the path doesn't exist. In any case, you can check if the include dir exists before adding it.

Reply:
Yes, the compilr would not mind if the directory does not exists. I have tried to find out how to get that prefix from `gcc` but I have not found anythign better than `gcc -### | grep configure | grep -- --prefix`. I would not expect that to always be set.

Reply:
Yeah, this is going to be a pain :)

I guess nix and gentoo prefix and friends don't use multiarch, exactly cause they have one package per prefix... but who knows they do support a single gcc that can target `gcc -m64 / -m32 / -mx32` on x86... :weary: 

And if they do, then glibc likely lives in a different prefix from gcc, so looking at gcc's prefix isn't useful

Reply:
OK. I have not been able to find anythign in `gcc`, `cpp`, and friends. Also, the reported install prefix is actually wrong for non-system installed ones. How about we only add this if the directory `/usr/include/...` exists?

Reply:
What GCC does is this: it has a bunch of config files per platform, they set the variable `MULTILIB_OSDIRNAMES`, this variable ultimately ends up in the output of `gcc -dumpspecs`.

that variable is typically set like:

```
$(call if_multiarch,:x86_64-linux-gnu)
```

where `x86_64-linux-gnu` is a hard-coded triplet, which `gcc -print-multiarch` prints.

Now, the `if_multiarch` make function is defined like this:

```make
if_multiarch = $(if $(wildcard $(shell echo $(BUILD_SYSTEM_HEADER_DIR))/../../usr/lib/*/crti.o),$(1))
```

so... it looks for glibc startup files relative to something, the relevant thing being that `*`, it's in a subdir of lib, not directly in lib.

---

long story short: maybe it's an idea to just let gcc compile helloworld.c in -###, and figure out the `crti.o` directory?

On Ubuntu, that is:

```
/usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu/crti.o
```

or 

```
/usr/lib/x86_64-linux-gnu/crti.o
```

If the dirname of crti.o isn't lib or lib64, or matches a triplet... use that?

Reply:
anyways, everything is awful here :weary:

Reply:
Hmm, nevermind, I don't think that works either, cause `if_multiarch` is only defined like that when multiarch is autodetected by GCC; debian / ubuntu just set the flag `--enable-multiarch` explicitly.

So in the end you still have to figure out how gcc determines the `/usr/include` dir, I guess.

Reply:
This could work:

```
$ `gcc -print-prog-name=cc1` -imultiarch `gcc -print-multiarch` -v
ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/12/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/12/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/12/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
```

Reply:
I just saw your last comment. So can we agree on this?
```
`gcc -print-prog-name=cc1` -imultiarch `gcc -print-multiarch` -v /dev/null 2>&1 | grep -v "nonexistent" | grep `gcc -print-multiarch`$
```